cmd/eval_test.go

```diff
@@ -1543,3 +1544,150 @@ func TestBundleWithStrictFlag(t *testing.T) {
 	}
 
 }
+
+func TestEvalDiscardOutput(t *testing.T) {
+	query := "1*2+3"
+	tests := []struct {
+		format, expected string
+	}{
+		{
+			format: evalDiscardOutput,
+			expected: `{
+  "result": "discarded"
+}
+`},
+	}
+	params := newEvalCommandParams()
+	err := params.outputFormat.Set(evalDiscardOutput)
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	var buf bytes.Buffer
+	_, err = eval([]string{query}, params, &buf)
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	if actual := buf.String(); actual != tests[0].expected {
+		t.Errorf("expected output %q\ngot %q", tests[0].expected, actual)
+	}
+}
+
+func TestEvalDiscardNILOutput(t *testing.T) {
+	query := "1/0"
+	tests := []struct {
+		format, expected string
+	}{
+		{
+			format: evalDiscardOutput,
+			expected: `{}
+`},
+	}
+	params := newEvalCommandParams()
+	err := params.outputFormat.Set(evalDiscardOutput)
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	var buf bytes.Buffer
+	_, err = eval([]string{query}, params, &buf)
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	if actual := buf.String(); actual != tests[0].expected {
+		t.Errorf("expected output %q\ngot %q", tests[0].expected, actual)
+	}
+}
+
+func TestEvalDiscardShowBuiltinErrors(t *testing.T) {
+	query := "1/0"
+	tests := []struct {
+		format, expected string
+	}{
+		{
+			format: evalDiscardOutput,
+			expected: `{
+  "errors": [
+    {
+      "code": "eval_builtin_error",
+      "location": {
+        "col": 1,
+        "file": "",
+        "row": 1
+      },
+      "message": "div: divide by zero"
+    }
+  ]
+}
+`},
+	}
+	params := newEvalCommandParams()
+	err := params.outputFormat.Set(evalDiscardOutput)
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	params.showBuiltinErrors = true
+	var buf bytes.Buffer
+	_, err = eval([]string{query}, params, &buf)
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	if actual := buf.String(); actual != tests[0].expected {
+		t.Errorf("expected output %q\ngot %q", tests[0].expected, actual)
+	}
+}
+
+func TestEvalDiscardProfilerOutput(t *testing.T) {
```

Comment:
suggestion: For this test, I'd be happy to just validate that the profile field is still present and the result is "discarded". I don't think we need to make the assertions about the profile output format.